src/extension.ts

```diff
@@ -263,10 +266,10 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
         traceLog(`Python extension loading`);
         await initializePython(context.subscriptions);
         traceLog(`Python extension loaded`);
+        return; // The `onDidChangePythonInterpreter` event will trigger the server start.
       }
-    } else {
-      await runServer();
     }
+    await runServer();
   });
```

Comment:
This was a bug that would hang the extension when `ruff.interpreter` is provided and the workspace is trusted.